### DISCUSSION TITLE
Не работает с мгтс

### DISCUSSION THREAD

**Discussion by JacobMaltsev** [hufrea/byedpi#35] [General]
Какие параметры я должен поменять, чтобы приложение заработало с мгтс? Если с GoodByeDPI это доп. аргументы "-9 -e1 -f1", где последние два аргумента параметры http и https, то как здесь?

**Comment by hufrea** [comment]
А на чем у вас запущен byedpi? Linux/Windows?

**Reply by JacobMaltsev** [reply]
я говорю про андроид

**Comment by lufog** [comment]
GoodbyeDPI `-9` разворачивается в `-f 2 -e 2 --wrong-seq --wrong-chksum --reverse-frag --max-payload -q`
| Key | Description |
|------------|------------|
| -f \<value\> | set HTTP fragmentation to value |
| -e \<value\> | set HTTPS fragmentation to value |
| --wrong-seq | activate Fake Request Mode and send it with TCP SEQ/ACK in the past. |
| --wrong-chksum | activate Fake Request Mode and send it with incorrect TCP checksum. |
| --reverse-frag | fragment (split) the packets just as --native-frag, but send them in the reversed order. Works with the websites which could not handle segmented HTTPS TLS ClientHello (because they receive the TCP flow "combined"). |
| --max-payload [value] | packets with TCP payload data more than [value] won't be processed. Use this option to reduce CPU usage by skipping huge amount of data (like file transfers) in already established sessions. May skip some huge HTTP requests from being processed. Default (if set): --max-payload 1200. |
| -q | block QUIC/HTTP3 |

Не уверен как это спроэцировать на ByeDPI. Могу только предположить, что-то типа:
Linux
```
--disorder 1 --fake -1 --md5sig
```
Windows
```
--disorder 1 --fake -1
```

**Comment by zimbabwe303** [comment]
Я не на МГТС, но у меня работает `--disorder 1 --auto --fake -1 --tlsrec 3+h` . Заходит даже на rutracker.

EDIT: заменил `--split -1` на `--disorder 1` (можно и `--split 1`), иначе youtube по-прежнему тормозил.

**Reply by nikitadan4pi** [reply]
на рутрекер у меня и так заходит, в вот Ютуб грузит картинки и названия, но ролики и описания нет

**Comment by JacobMaltsev** [comment]
Дак я использую приложение для андроида, а не винду, мне непонятно, какие тут значения выставлять, то, что вы пишете, к сожалению, даёт мне довольно мало информации.

**Comment by lufog** [comment]
@JacobMaltsev насколько я понимаю, приложение для андроида, ~не поддерживает указание нескольких методов одновременно. Так что нужно выбрать что-то одно.~ С версии 1.0.2 поддерживает.

Для начала можно попробовать
---

`Desync method: disorder`
`Split position: 1` (со смещением можно поэкспериментировать, например указать отрицательное)

Можно попробовать также включить: 
`Split at host`
`Host mixed case` (очень редко помогает)
`Host remove spaces` (очень редко помогает)

Более нестабильные опции, чаще помогают, но могут сломать сертификаты на сайтах. Например, у меня ломает сертификат от ddos-guard.net.

`Split TLS record`
`TLS record split posotion 1`
`Split TLS record at SNI`

Если не сработало попробовать
---

`Desync method: fake`
`Split position: 1` (со смещением можно поэкспериментировать, например указать отрицательное)
`TTL of fake packets: 10` (проверять и уменьшать цифру, пока сайт не заработает)

Еще я бы советовал перезапускать хром после каждой смены настроек, и переподключения, чтобы исключить его затупы.

**Comment by At3m1** [comment]
У меня получилось завести Byedpi на андроид на Smart TV. Провайдер мгтс. Вот настройки.

DNS
8.8.8.8

Listen Adress
127.0.0.1

Port
1080

Maximum number of connections
512

Buffer size
16384

Default ttl
0

No domain
без галочки

Desync only https and tls
без галочки

Desync method
Fake

Split Position
1

Split and host
без галочки

TTL of fake packets
7

Sni of fake packet
www.iana.org

Oob data
a

Host mixed case
без галочки

Domain mixed case
без галочки

Host remove spaces
без галочки

Split tls record
без галочки

tls record split position
1

split tls record at sni
без галочки

работает в режиме VPN. первый раз ютуб может долго заводиться (примерно минуту)

**Reply by TofertFoxy** [reply]
Лично мне, тоже на МГТС сижу, так ещё и с их родным роутером, помогло следующее. Нащупал сам, но надеюсь и у других будет работать (сразу оговорюсь, я не совсем понимаю как это работает, но мне достаточно просто факта обхода замедления ютуба, не более, с чем мой способ конкретно у меня справляется) 

Попробуй поставить "SNI of fake packet" как yandex.ru

В desync method выбери none

В поле "Split at host" поставь галочку

**Reply by zemuro** [reply]
У меня не сработало, но сработали такие настройки: desync method: split, галки на desync https, desync UDP, split position: 2, split at host, split TLS record, TLS split position: 1

Подключается, видео проигрывает, но перед началом воспроизведения задержка в несколько секунд. 🤷

**Reply by zemuro** [reply]
А теперь снова не работает

**Comment by apollohot** [comment]
А есть ли решение для проводного МТС мск через настроеного MikroTik? Не работает со строкой по умолчанию: --disorder 1 --auto=torst --tlsrec 1+s --debug 1

С настройками по умолчанию через MikroTik на провайдере Beeline мск работает прекрасно всё.

**Comment by BeragesFrares** [comment]
Приложение на Андроид поддерживает указание целиком командной строки. У меня для МГТС 14.10.2024 работает со строкой

`-s1 -q1 -Y -Ar -s5 -o1+s -At -f-1 -r1+s -As -s1 -o1 +s -s-1 -An`

(она взята с ветки на https://ntc.party)

Скорее всего ByeDpi для linux должен идентично работать. А вот для Windows не так похоже, как я упомянул в другой ветке, параметр `-Y` не поддерживается под Windows. Убирание из Windows не привело к позитивным результатам (ff, chrome - youtube). Попытка убрать его из Android-а (чтобы понять насколько важна) приводила к частичным позитивным результатам (правда это было приложение NewPipe и тесты с youtube по ее логике). Возврат параметра `-Y` вернул результаты к очень хорошим. Я подозреваю, что в этой строке закодирован сложный алгоритм действий, так как как видно, что используются несколько раз параметр `-s` (split)

**Reply by zhk-kk** [reply]
Перестало работать на МГТС в мск (для ютуба)

**Reply by Frozen-Brocolis** [reply]
Работает с МТС и МГТС (МСК) если поменять DNS на 94.140.14.14 еще можно отключить IPv6 , видео грузится побыстрее но я не уверен.

**Comment by Frozen-Brocolis** [comment]
Москва МГТС и МТС 
--proto=udp --pf=443 --fake-data=':\xC2\x00\x00\x00\x01\x14\x2E\xE3\xE3\x5F\x6B\xBB\x23\xA8\xE6\x5D\xA9\x78\x21\xCF\xC2\x72\x4C\x8F\xC4\x5E\x14\x00\x00\x00\x00\xC5\x00\x00\x00\x00\x4C\x00\xA7\x00\x00\x00\x00\x00\x00\x44\x00\x00\x80\x00\x00\x00\x0D\xFC\xFA\x1D\xCD\x73\xBA\x2A\x90\x93\xB3\xEE\xF7\x43\xC5\x85\xDA\xFF\x45\x3C\x00\x00\x00\x00\x00\x00\x7C\x00\x9B\x00\xF6\x00\x00\xDD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x59\xA8\xE4\x00\x00\x00\x00\x00\x00\x00\x00\x7B\x00\x0F\x00\x00\x00\x48\x4E\x00\x00\x00\x06\xF3\x00\x00\x00\x00\xD9\x5A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' --udp-fake=10 --auto=none -s1 -o1

DNS 8.8.8.8

самый быстрый способ из всех что видел

**Reply by boss5588** [reply]
На телефоне отлично работает. Правда, на Сони андроид ТВ не работает при вайфае мтс

**Reply by Amderos** [reply]
Середина февраля 25 года, работает.
Спасибо!

**Reply by ARBlazeAG** [reply]
Апрель 22, перестало работать!

**Comment by trevarj** [comment]
этот конфиг тоже работает с мгтс (youtube/discord chat-only): `-o1 -o25+s -T3 -At --tlsrec 1+s` 
из https://github.com/hufrea/byedpi/discussions/155?sort=new#discussioncomment-11865692

**Reply by boss5588** [reply]
На телевизоре на смарттюбе не работает

**Reply by Nagatoro03** [reply]
На телевизор была другая командная строка, поищи выше в топике

**Reply by ARBlazeAG** [reply]
Апрель 22.04.2025 работает

**Comment by AntonyBenjiro** [comment]
https://github.com/hufrea/byedpi/issues/261#issuecomment-2634432690

> Внезапно бОльшая часть сайтов перестала работать. Гитхаб, твич и вконтакте открываются. Ютуб, ya.ru, гугл - нет. спидтест открывается частично.
> 
> МГТС MacOS M1 Sonoma 14.6.1 byedpi v14.6, v16.4
> 
> Ошибка в терминале: `recv: Network is unreachable`
> 
> если отключаю прокси, начинают работать "разрешенные сайты" (гугл, яндекс и т.д.)
> 
> Пробовал разные комбинации запуска но еще утром хватало такой: `./ciadpi -d1 -o25+s -Hwhitelist`

**Comment by Nagatoro03** [comment]
Есть обновление командной строки для смарт тв на андроиде, Москва МТС? Я про ByeByeDpi

**Reply by Nagatoro03** [reply]
МТС, Москва. ByeByeDPI 1.3.2 версия, днс 8.8.8.8, смарт тв андроид, ютуб работает. Не путать с смартюб. Смотрю ютуб через оф приложение. 
-d2+s -s 50+s -O 2 -a 10 -As -f20 -r2+s -T2 -At -d2 -s0+s -s5+s -s10+s -s15+s -s25+s -s35+s -s50+s -s60+s -An -n youtube.com -d2:5:2+h -f-3 -r2+sm -o2 -o50+s -r2+s -f-4 -r0+hm -t3

**Comment by Alexei-Kassian** [comment]
По-моему, все из предложенных выше наборов ключей перестали работать с МГТС Москва (для ютуба, по крайней мере).

**Reply by Frozen-Brocolis** [reply]
--proto=udp --pf=443 --fake-data=':\xC2\x00\x00\x00\x01\x14\x2E\xE3\xE3\x5F\x6B\xBB\x23\xA8\xE6\x5D\xA9\x78\x21\xCF\xC2\x72\x4C\x8F\xC4\x5E\x14\x00\x00\x00\x00\xC5\x00\x00\x00\x00\x4C\x00\xA7\x00\x00\x00\x00\x00\x00\x44\x00\x00\x80\x00\x00\x00\x0D\xFC\xFA\x1D\xCD\x73\xBA\x2A\x90\x93\xB3\xEE\xF7\x43\xC5\x85\xDA\xFF\x45\x3C\x00\x00\x00\x00\x00\x00\x7C\x00\x9B\x00\xF6\x00\x00\xDD\x00\x00\x00\x00\x00\x00\x00\x00\x00\x59\xA8\xE4\x00\x00\x00\x00\x00\x00\x00\x00\x7B\x00\x0F\x00\x00\x00\x48\x4E\x00\x00\x00\x06\xF3\x00\x00\x00\x00\xD9\x5A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' --udp-fake=10 --auto=none -s1 -o1

8.8.8.8
IPv6 включен 

У меня работает - МГТС Москва
